doc/mgmt/Management Framework.md

```diff
@@ -0,0 +1,2029 @@
+* [7 Serviceability and Debug](#7-serviceability-and-debug)
+* [8 Warm Boot Support](#8-warm-boot-support)
+* [9 Scalability](#9-scalability)
+* [10 Unit Test](#10-unit-test)
+
+
+## List of Tables
+
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+## Revision
+
+| Rev |     Date    |       Author            | Change Description                |
+|:---:|:-----------:|:-----------------------:|-----------------------------------|
+| 0.1 | 06/13/2019  | Anand Kumar Subramanian | Initial version                   |
+| 0.2 | 07/05/2019  | Prabhu Sreenivasan      | Added gNMI, CLI content from DELL |
+| 0.3 | 08/05/2019  | Senthil Kumar Ganesan   | Updated gNMI content          |
+| 0.4 | 08/07/2019  | Arun Barboza            | Clarifications on Table CAS   |
+| 0.5 | 08/07/2019  | Anand Kumar Subramanian | Translib Subscribe support    |
+| 0.6 | 08/08/2019  | Kwangsuk Kim            | Updated Developer Workflow and CLI sections |
+| 0.7 | 08/09/2019  | Partha Dutta            | Updated Basic Approach under Design Overview |
+| 0.8 | 08/15/2019  | Anand Kumar Subramanian | Addressed review comments     |
+| 0.9 | 08/19/2019  | Partha Dutta            | Addressed review comments related to CVL    |
+| 0.10 | 09/25/2019  | Kwangsuk Kim           | Updated Transformer section |
+| 0.11 | 09/30/2019  | Partha Dutta           | Updated as per SONiC YANG guideline    |
+
+## About this Manual
+
+This document provides general information about the Management framework feature implementation in SONiC.
+
+## Scope
+
+This document describes the high level design of Management framework feature.
+
+## Definition/Abbreviation
+
+### Table 1: Abbreviations
+
+| **Term**                 | **Meaning**                         |
+|--------------------------|-------------------------------------|
+| CVL                      | Config Validation Library           |
+| NBI                      | North Bound Interface               |
+| ABNF                     | Augmented Backus-Naur Form          |
+| YANG                     | Yet Another Next Generation         |
+| JSON                     | Java Script Object Notation         |
+| XML                      | eXtensible Markup Language          |
+| gNMI                     | gRPC Network Management Interface   |
+| YGOT                     | YANG Go Tools                       |
+
+## 1 Feature Overview
+
+Management framework is a SONiC application which is responsible for providing various common North Bound Interfaces (NBIs) for the purposes of managing configuration and status on SONiC switches. The application manages coordination of NBI’s to provide a coherent way to validate, apply and show configuration.
+
+### 1.1 Requirements
+
+* Must provide support for:
+
+    1. Standard [YANG](https://tools.ietf.org/html/rfc7950) models (e.g. OpenConfig, IETF, IEEE)
+    2. Custom YANG models ([SONiC YANG](https://github.com/Azure/SONiC/blob/master/doc/mgmt/SONiC_YANG_Model_Guidelines.md))
```

Comment:
All of them must be supported? What if SONiC doesn't have support for all the models?

Reply:
If SONiC does not have support, those models won't be implemented. Also if SONiC has partial support for a model, other non-supported objects will be marked as "not-supported" through deviation.

Reply:
So we create another model based on standard model.
Do we need another non-standard model?

Reply:
With the SONiC yang model, the conversion is fairly straightforward as it is 1-1 with the ABNF schema in Redis. Because of this many in the work group wanted support for SONiC yang. With this architecture we will support both the models.

Reply:
So we should remove standard model here, because SONiC doesn't support it. Right?

Reply:
SONiC supports both. We have equal number in the work group asking support for standard models as well. So our architecture supports both.

Reply:
I really want to get the idea.
I don't know why, but I'm pretty sure that SONiC doesn't have support for all settings for the standard model. And I also sure that the standard models doesn't have support for all SONiC settings. It's a reason why you want to use SONiC models (otherwise why do we need them? Let's use the standard?). But as soon as we want support the standard models, we must change them to make compatible with our requirements making them non-standard.
So why do want to support standard models, making them non-standard?

Reply:
For a given feature we may have support for both Standard models as well as SONiC models. When standard models does not have support, we will provide standard like yang models. This is to support users who want only standard models/standard like models and users who want SONiC only models.